### ISSUE TITLE
Does a Thunk really need an explicit call to fail

### ISSUE THREAD

**Issue by kuhmuh** [ctrlplusb/easy-peasy#604]
Hello,

I enjoy Easy Peasy very much and just want to migrate to v4 and have some failing tests. I stumbled on [this comment](https://github.com/ctrlplusb/easy-peasy/issues/568#issuecomment-710221567) and want clarification about this, because it is not yet documented. In v3 a failed thunk (to be exact: at least a thunk that returns a rejected `Promise`) triggers a listener with a `failType`. In v4 it does not. The comment suggests to call `fail` helper method if a thunk fails. Is this really necessary for every "failed/rejected" thunk?
Thanks for clarification.

Best regards
Stefan

**Comment by LuisOsta** [comment]
Yeah I think at least adding more documentation about how error handling works in thunks would be great

**Comment by macrozone** [comment]
i have the same question. I hesitate upgrading to v4 because i fear that it now silently swallows errors in thunks.

I don't use `actionOn` and `thunkOn` (tbh. i dont understand the use case for that), am i still affected by this change?
